src/main/java/seedu/address/commons/util/ArrayListMap.java

```diff
@@ -62,7 +63,7 @@ public V put(K key, V value) {
         if (index == -1) {
             keys.add(key);
             vals.add(value);
-            index = keys.indexOf(key);
+            return null;
```

Comment:
This is the correct fix for the bug -- `put` is supposed to return the previous value, or `null` if there is none.